src/main/java/org/terracotta/statistics/extended/CompoundOperationImpl.java

```diff
@@ -235,16 +228,8 @@ public boolean expire(long expiryTime) {
         // Not using && on purpose here. expire() has a side-effect. We want to make sure it's called (no short-circuit evaluation)
         expired &= o.expire(expiryTime);
       }
-      for (Iterator<ResultImpl<T>> it = compounds.values().iterator(); it.hasNext(); ) {
-        if (it.next().expire(expiryTime)) {
-          it.remove();
-        }
-      }
-      for (Iterator<ExpiringSampledStatistic<Double>> it = ratios.values().iterator(); it.hasNext(); ) {
-        if (it.next().expire(expiryTime)) {
-          it.remove();
-        }
-      }
+      compounds.values().removeIf(tResult -> tResult.expire(expiryTime));
+      ratios.values().removeIf(doubleExpiringSampledStatistic -> doubleExpiringSampledStatistic.expire(expiryTime));
```

Comment:
I imagine this choice was IntelliJ's - could we switch to a shorter parameter name?  Could probably get away with just `r`.

Reply:
IntelliJ indeed. I've put `result` and `statistic`